awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/OtlpUdpSpanExporter.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.opentelemetry.javaagent.providers;
+
+import io.opentelemetry.exporter.internal.otlp.traces.TraceRequestMarshaler;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public class OtlpUdpSpanExporter implements SpanExporter {
+
+  private static final Logger logger = Logger.getLogger(OtlpUdpSpanExporter.class.getName());
+  private static final String PROTOCOL_HEADER = "{\"format\": \"json\", \"version\": 1}";
+  private static final char PROTOCOL_DELIMITER = '\n';
+  private static final String FORMAT_OTEL_SAMPLED_TRACES_BINARY_PREFIX = "T1S";
+  private static final String FORMAT_OTEL_UNSAMPLED_TRACES_BINARY_PREFIX = "T1U";
+
+  private final AtomicBoolean isShutdown = new AtomicBoolean();
+
+  private final UdpSender sender;
+  private final boolean sampled;
```

Comment:
It's really confusing why sampled decision is at exporter level, not span level. Can you add comment on how it is used?

Reply:
I agree that it is confusing and unfortunately I don't have a great explanation for this. 
When on Lambda, the entire batch of spans will either be sampled or unsampled. Since Lambda execution is short lived, this sample information can be set when the OtlpUdpSpanExporter is being initialized once, used later during export.

For reference: https://github.com/aws-observability/aws-otel-python-instrumentation/pull/247

Reply:
Thanks for the reference. I see the decision is made on span in the Python version. The traffic is diverged at the leading span processor. Will we have the same solution in Java? Have we evaluated if the same solution can be applied?